babs/babs.py

```diff
@@ -1012,119 +1014,32 @@ def babs_submit(self, count=1, df_job_specified=None):
         lock_path = self.job_status_path_abs + ".lock"
         lock = FileLock(lock_path)
 
-        j_count = 0
-
         try:
             with lock.acquire(timeout=5):  # lock the file, i.e., lock job status df
                 df_job = read_job_status_csv(self.job_status_path_abs)
+                # getting list of jobs indices to submit (based either on df_job_specified or count):
+                job_ind_list = prepare_job_ind_list(df_job, df_job_specified, count, self.type_session)
                 df_job_updated = df_job.copy()
+                for i_progress, i_job in enumerate(job_ind_list):
```

Comment:
Before this, not sure if you need to add a sanity check of "if `job_ind_list` is empty"? i.e., when there is no job to submit.

Reply:
i think it should be fine, it will just do nothing